### PR TITLE
fix(TDI-40055): GoogleDrive launch browser in connection's wizard

### PR DESCRIPTION
**Pull request by undx**
**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TDI-40055

related 
https://jira.talendforge.org/browse/TDI-39455
https://jira.talendforge.org/browse/TUP-19485

**What is the new behavior?**

Fixed problem when trying to launch user consent when we are in the connection's wizard.

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [X] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1721/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1721/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-40055_GoogleDriveLaunchBrowserWizard)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1724/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1724/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-40055_GoogleDriveLaunchBrowserWizard)

File: components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveRuntime.java [hunk 1]
```diff
@@ -57,25 +58,40 @@
...
     private GoogleDriveUtils utils;
 
     private NetHttpTransport httpTransport;
 
-    protected GoogleDriveProvideConnectionProperties properties;
-
-    private transient static final Logger LOG = LoggerFactory.getLogger(GoogleDriveRuntime.class);
+    public static String getStudioName(final String inputString) {
+        if (inputString == null || inputString.isEmpty()) {
+            return inputString;
```
Comment (ypiel-talend) [inline components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveRuntime.java]: Null is a valid return value ?

File: components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveRuntime.java [hunk 2]
```diff
@@ -57,25 +58,40 @@
...
+        }
 
-    protected static final I18nMessages messages = GlobalI18N.getI18nMessageProvider()
-            .getI18nMessages(GoogleDriveSourceOrSink.class);
+        StringBuilder outputString = new StringBuilder();
+        for (int i = 0; i < inputString.length(); i++) {
+            Character c = inputString.charAt(i);
+            outputString.append(Character.isUpperCase(c) && i > 0 ? "_" + c : c); //$NON-NLS-1$
```
Comment (ypiel-talend) [inline components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveRuntime.java]: when '"_"+c'  c shouldn't be converted in lowercase ?

File: components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/client/AuthorizationCodeInstalledAppTalend.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+public class AuthorizationCodeInstalledAppTalend extends AuthorizationCodeInstalledApp {
+
+    private transient static final Logger LOG = LoggerFactory.getLogger(AuthorizationCodeInstalledAppTalend.class);
+
+    public AuthorizationCodeInstalledAppTalend(AuthorizationCodeFlow flow, VerificationCodeReceiver receiver) {
+        super(flow, receiver);
+    }
+
+    public static void browseWithProgramLauncher(String url) {
+        Preconditions.checkNotNull(url);
```
Comment (ypiel-talend) [inline components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/client/AuthorizationCodeInstalledAppTalend.java]: NPE is thrown is url == null that's it ?

File: components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveRuntime.java (hunk 2 above)
Comment (undx) [reply components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveRuntime.java]: No, all caps in studio.

File: components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/client/AuthorizationCodeInstalledAppTalend.java (hunk 3 above)
Comment (undx) [reply components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/client/AuthorizationCodeInstalledAppTalend.java]: Yes, It's inherited directly from google API's code.

File: components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveRuntime.java (hunk 1 above)
Comment (undx) [reply components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveRuntime.java]: Nothing to translate in that case. Null or empty string won't be stored in container. Should maybe return an empty string in all case...

File: components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveRuntime.java (hunk 2 above)
Comment (ypiel-talend) [reply components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveRuntime.java]: ok

File: components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/client/AuthorizationCodeInstalledAppTalend.java (hunk 3 above)
Comment (ypiel-talend) [reply components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/client/AuthorizationCodeInstalledAppTalend.java]: ok

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1725/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1725/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-40055_GoogleDriveLaunchBrowserWizard)

**Pull request merged**